soroban-sdk/src/env.rs

```diff
@@ -244,6 +244,28 @@ impl Env {
     pub fn log_value<V: IntoVal<Env, RawVal>>(&self, v: V) {
         internal::Env::log_value(self, v.into_val(self));
     }
+
+    #[doc(hidden)]
+    pub fn get_ledger_version(&self) -> u32 {
+        internal::Env::get_ledger_version(self).try_into().unwrap()
+    }
+
+    #[doc(hidden)]
+    pub fn get_ledger_sequence(&self) -> u32 {
+        internal::Env::get_ledger_sequence(self).try_into().unwrap()
+    }
+
+    #[doc(hidden)]
+    pub fn get_ledger_timestamp(&self) -> u64 {
+        let obj = internal::Env::get_ledger_timestamp(self);
+        internal::Env::obj_to_u64(self, obj)
+    }
+
+    #[doc(hidden)]
+    pub fn get_ledger_network_id(&self) -> Bytes {
+        let bin_obj = internal::Env::get_ledger_network_id(self);
+        bin_obj.in_env(self).try_into().unwrap()
+    }
```

Comment:
Could these functions be moved into a `Ledger` struct, using the same pattern as `ContractData`? 

> ```rust
> impl Env {
>     pub fn current_ledger(&self) -> Ledger
> ```
> 
> ```rust
> impl Ledger {
>     pub fn number(&self) -> u32;
>     pub fn close_time(&self) -> ?;
> ```

Ref: https://github.com/stellar/rs-soroban-env/issues/312#issuecomment-1208631832